### DISCUSSION TITLE
Tab support in macOS non-native fullscreen and quick terminal

### DISCUSSION THREAD

**Discussion by jimeh** [ghostty-org/ghostty#3629] [Feature Requests, Ideas]
I understand there are technical limitations with adding tab support to both quick terminal and macOS non-native full screen windows. Despite that, it would still be nice, ideally with the tab-bar being hidden by default.

What I'd personally want is a non-native fullscreen quick terminal, with tab support. It's essentially how I use iTerm right now, with a global hotkey for a fullscreen terminal, within which I'll use tmux for all terminal multi-plexing, but separate tabs for local my system, remote system 1 (with tmux), remote system 2 (with tmux), etc., while switching between tabs with cmd+<num> and cmd+] and cmd+[.

**Comment by 00-kat** [comment]
My understanding is that the native APIs used simply don't work in them; it was mentioned in https://ghostty.org/docs/help/macos-tiling-wms (see the blue Note section) that a custom tabbing solution is planned (not anytime soon though).

**Comment by vinayak-mehta** [comment]
Any idea how I could open the terminal maximized like this? 

<img width="1512" alt="image" src="https://github.com/user-attachments/assets/ade87911-a529-4e61-a719-fa7be2063cce" />

Even though tab support would be nice in the non-native fullscreen, I can live with this workaround but not sure what shortcuts to configure to maximize ghostty like this. I'm basically tiling the ghostty window by selecting the fill as full (the first option in Fill & Arrange):

<img width="873" alt="image" src="https://github.com/user-attachments/assets/882d5f47-2104-4463-91ef-d0721f81f803" />

With iterm, I could use control+option+enter to achieve this by default, but can't figure out how to do the same with ghostty.

**Reply by vinayak-mehta** [reply]
I'm trying to fill using a keyboard shortcut I've defined at the macOS level:
<img width="888" alt="image" src="https://github.com/user-attachments/assets/0720106d-f65e-4df4-8161-fbd6e3ed1ce1" />
<img width="447" alt="image" src="https://github.com/user-attachments/assets/71cee859-3661-49b0-91bb-f56e487626da" />

But when I try to use it with ghostty I just get these characters on the terminal instead:
<img width="809" alt="image" src="https://github.com/user-attachments/assets/4192d6c1-6923-485c-a10e-2b5df211e829" />

**Reply by vinayak-mehta** [reply]
I've figured out a way with the height and width config settings but would love to get my keyboard shortcuts working with ghostty:

```
window-height = 1000
window-width = 1000
```

**Reply by jimeh** [reply]
While not Ghostty specific, I use Hammerspoon for window management, and have a bunch of custom global keybindings for various fixed sizes/positions and movement/resizing operations. One of those is maximize.

If you're up for dabbling with Hammerspoon a bit, my window management config is here: https://github.com/jimeh/dotfiles/blob/bb538ee33b2b587b0780c9cf363456c08fa4d032/hammerspoon/window_management.lua

**Reply by edrozenberg** [reply]
Another great wm utility is Moom, lots of features especially their new v4 (download only, not in App Store). Works very well for me but haven't tried it with Ghostty specifically.  https://manytricks.com/moom/

**Comment by 00-kat** [comment]
xref #2888 and #9857, which have had a lot more attention.
